### PR TITLE
Fix the WSL/Ubuntu/GCC-9 TPL builds

### PR DESCRIPTION
**Pull request by jamiebramwell**
For some reason when updating spack, the `-ldl -lz` link flags got moved to the front of the link line for netcdf. I don't like this fix, so I am open to any other suggestions.

### CODE REVIEW

File: scripts/spack/packages/netcdf-c/package.py [hunk 1]
```diff
@@ -184,6 +184,11 @@ def configure_args(self):
         if hdf5_hl.satisfies('~shared'):
             libs.append(hdf5_hl.libs.link_flags)
 
+            # SERAC EDIT BEGIN
+            # WSL gcc 9 needs ldl and lz at the end of the link line
+            libs.append('-ldl -lz')
```
Comment (white238) [inline scripts/spack/packages/netcdf-c/package.py:189]: I can't even find where these were coming from hdf5 before.. @becker33 do you have an idea why this changed in the newest Spack?  This worked prior to updating to a newer develop.

Comment (white238) [reply scripts/spack/packages/netcdf-c/package.py:189]: So I don't forget this before I get to it.  Per @becker33, we should fix hdf5 to properly add `-ldl` to its libs when `~shared`.   I assume `-lz` is missing as well and should be added.

We think this was accidentally working before and the ordering changed and this issue was exposed. 

There is a fix I put into mfem for this as well that should be removed if this is fixed properly.

Comment (joshessman-llnl) [reply scripts/spack/packages/netcdf-c/package.py:189]: I have also run into issues where liblua is missing `-ldl` on WSL.  I imagine this is probably related.

**Comment by jamiebramwell** [comment]
Should we merge this as a stopgap fix until https://github.com/LLNL/serac/issues/324 gets resolved?

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
> Should we merge this as a stopgap fix until #324 gets resolved?

Yea I'll try to get to it this week.

**Comment by jamiebramwell** [comment]
@joshessman-llnl could you take a look at this?

**Review by joshessman-llnl — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM

**Pull request merged**